code/modules/modular_computers/networking/machinery/telecomms.dm

```diff
@@ -0,0 +1,301 @@
+		channel_data["channel_colour"] = channel_datum.color || COLOR_BLACK
+		if(channel_datum.secured)
+			channel_data["channel_access"] = jointext(channel_datum.secured, " ")
+		else
+			channel_data["channel_access"] = "PUBLIC"
+		channel_data["channel_ref"] =    "\ref[channel_datum]"
+		data_channels += list(channel_data)
+	data["channels"] = data_channels
+	ui = SSnano.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if (!ui)
+		ui = new(user, src, ui_key, "telecomms_hub.tmpl", "Telecommunications Hub")
+		ui.set_initial_data(data)
+		ui.open()
+		ui.set_auto_update(1)
+
+/obj/machinery/network/telecomms_hub/interface_interact(mob/user)
+	ui_interact(user)
+	return TRUE
+
+/obj/machinery/network/telecomms_hub/OnTopic(mob/user, href_list, datum/topic_state/state)
+	. = ..()
+	if(!.)
+		if(href_list["channel"])
+			var/datum/comms_channel/channel_datum = locate(href_list["channel"])
+			if(!istype(channel_datum) || !(channel_datum in channels))
+				. = TOPIC_HANDLED
+			else
+
+				if(href_list["change_name"])
+					var/new_name = input(user, "Enter a new channel name.", "Channel Configuration", channel_datum.name) as text|null
+					new_name = sanitize(new_name, MAX_NAME_LEN)
+					if(new_name != channel_datum.name && new_name && CanPhysicallyInteract(user))
+						channel_datum.name = new_name
+						. = TOPIC_REFRESH
+
+				if(href_list["change_key"])
+					var/new_key = input(user, "Enter a new key character.", "Channel Configuration", channel_datum.key) as text|null
+					new_key = sanitize(new_key, 1)
+					if(new_key != channel_datum.key && new_key && CanPhysicallyInteract(user))
+						if(new_key in global.special_channel_keys)
+							to_chat(user, SPAN_WARNING("You may not use any of the following keys: [english_list(global.special_channel_keys)]."))
+							return
+						channel_datum.key = new_key
+						rebuild_channels()
+						. = TOPIC_REFRESH
+
+				if(href_list["change_freq"])
+
+					var/lower_freq
+					var/upper_freq
+					if(access_keycard_auth in user.GetAccess())
+						lower_freq = RADIO_LOW_FREQ
+						upper_freq = RADIO_HIGH_FREQ
+					else
+						lower_freq = PUBLIC_LOW_FREQ
+						upper_freq = PUBLIC_HIGH_FREQ
+
+					var/new_freq = input(user, "Enter a new frequency beween [lower_freq] and [upper_freq].", "Channel Configuration", channel_datum.frequency) as num
+
+					if(access_keycard_auth in user.GetAccess())
+						lower_freq = RADIO_LOW_FREQ
+						upper_freq = RADIO_HIGH_FREQ
+					else
+						lower_freq = PUBLIC_LOW_FREQ
+						upper_freq = PUBLIC_HIGH_FREQ
+					new_freq = Clamp(new_freq, lower_freq, upper_freq)
+
+					if(new_freq != channel_datum.frequency && new_freq && CanPhysicallyInteract(user))
+						channel_datum.frequency = new_freq
+						rebuild_channels()
+						. = TOPIC_REFRESH
+
+				if(href_list["change_colour"])
+					var/new_color = input(user, "Select a new display colour.", "Channel Configuration", channel_datum.color) as color|null
```

Comment:
letting people freely picking comms colors is not ideal imo, white / acid yellow colors etc.

Reply:
Is that necessarily a problem? It adds a dimension of fuckaroundability that existing tcomms lacks.

Reply:
I just hate freedom is all. Not a big thing.

Reply:
Made them pick from the tcomms colour list instead.